shm/test.cpp

```diff
@@ -0,0 +1,45 @@
+#include "cargo.hpp"
+#include "coordinates.hpp"
+#include "island.hpp"
+#include "map.hpp"
+#include "ship.hpp"
+#include "player.hpp"
+
+#include <vector>
+#include "gtest/gtest.h"
+
+
+class SHM_TestFixture: public ::testing::Test {
+public:
+    SHM_TestFixture( ) {
+        // initialization code here
+    }
+
+    void SetUp( ) {
+        // code here will execute just before the test ensues
+    }
+
+    void TearDown( ) {
+        // code here will be called just after the test completes
+        // ok to through exceptions from here if need be
+    }
+
+    ~SHM_TestFixture( )  {
+        // cleanup any pending stuff, but no exceptions allowed
+    }
+
+    // put in any custom data members that you need
+    Map testMap;
+};
+
+
+TEST_F(SHM_TestFixture, ShouldAlwaysPass) {
+
+    EXPECT_TRUE(true);
+}
+
+TEST_F(SHM_TestFixture, ShouldReturnNullPtrWhenAskedForIslandOutOfRange) {
+
+    Coordinates location(MAP_RND_DISTRIBUTION_MAX + 10, MAP_RND_DISTRIBUTION_MAX + 10);
```

Comment:
testMap doesn't use this location variable, does it?

Reply:
testMap is not using this variable. testMap has only 10 islands with coordinates between MAP_RND_DISTRIBUTION_MIN and MAP_RND_DISTRIBUTION_MAX. We want to check only if nullptr is returned if island is not found in map